{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Open Reading Frames & Regression\n",
    "\n",
    "**1(a)**\n",
    "\n",
    "``` r\n",
    "findStartCodons <- function(seq){\n",
    "  startcodons <- numeric(0)\n",
    "  k <- 1\n",
    "  for(i in 1:(length(seq)-5)){\n",
    "    if (seq[i] == \"a\" && seq[i + 1] == \"t\" && seq[i + 2] == \"g\") {\n",
    "      startcodons <- c(startcodons, k)\n",
    "    }\n",
    "    k <- k + 1\n",
    "  }\n",
    "  return(startcodons)\n",
    "}\n",
    "\n",
    "seq <- c(\"g\", \"t\", \"a\", \"a\", \"t\", \"g\", \"t\", \"a\", \"g\", \"t\", \"g\", \"a\", \"t\", \"t\", \"g\", \"t\", \"a\", \"g\")\n",
    "findStartCodons(seq)\n",
    "```\n",
    "\n",
    "    [1] 4\n",
    "\n",
    "**1(b)**\n",
    "\n",
    "``` r\n",
    "findStopCodons <- function(seq){\n",
    "  stopcodons <- numeric(0)\n",
    "  k <- 1\n",
    "  for(i in 1:(length(seq)-2)){\n",
    "    if((seq[i] == \"t\" && seq[i+1] == \"a\" && seq[i+2] == \"a\") ||\n",
    "      (seq[i] == \"t\" && seq[i+1] == \"g\" && seq[i+2] == \"a\") ||\n",
    "      (seq[i] == \"t\" && seq[i+1] == \"a\" && seq[i+2] == \"g\")){\n",
    "      stopcodons[k]<- i\n",
    "      k <- k+1\n",
    "    }\n",
    "  }\n",
    "  return(stopcodons)\n",
    "}\n",
    "\n",
    "seq<-c(\"g\", \"t\", \"a\", \"a\", \"t\", \"g\", \"t\", \"a\", \"g\", \"t\", \"g\", \"a\", \"t\", \"t\", \"g\", \"t\", \"a\", \"g\")\n",
    "findStopCodons(seq)\n",
    "```\n",
    "\n",
    "    [1]  2  7 10 16\n",
    "\n",
    "**1(c)**\n",
    "\n",
    "``` r\n",
    "findStopCodons <- function(seq){\n",
    "  stopcodons <- numeric(0)\n",
    "  k <- 1\n",
    "  for(i in 1:length(seq)){\n",
    "    if((seq[i] == \"t\" && seq[i+1] == \"a\" && seq[i+2] == \"a\") ||\n",
    "      (seq[i] == \"t\" && seq[i+1] == \"g\" && seq[i+2] == \"a\") ||\n",
    "      (seq[i] == \"t\" && seq[i+1] == \"a\" && seq[i+2] == \"g\")){\n",
    "      stopcodons[k]<- i\n",
    "      k <- k+1\n",
    "    }\n",
    "  }\n",
    "  return(stopcodons)\n",
    "}\n",
    "\n",
    "seq<-c(\"g\", \"t\", \"a\", \"a\", \"t\", \"g\", \"t\", \"a\", \"g\", \"t\", \"g\", \"a\", \"t\", \"t\", \"g\", \"t\", \"a\", \"g\")\n",
    "findStopCodons(seq)\n",
    "```\n",
    "\n",
    "    [1]  2  7 10 16\n",
    "\n",
    "<span style=\"color:blue\"> </span>\n",
    "\n",
    "**1(d)** <span style=\"color:blue\"> To ensure that the identified start\n",
    "codon is part of an open reading frame, we need to leave room for the\n",
    "stop codon to occur after the start codon. By iterating over\n",
    "1:length(seq)-5 in 1(a), we allow for at least six elements to be\n",
    "present after the start codon. This way, if a start codon is found,\n",
    "there is sufficient space to search for the subsequent stop codons. On\n",
    "the other hand, in 1(b) when search for stop codon, there is need to\n",
    "leave additional elements after the stop codon. The stop codons can\n",
    "occur immediately after the start codon or after any number of\n",
    "nucleotides within the ORF. Therefore, the loop iterates over\n",
    "1:length(seq)-2 to cover all possible positions of the stop codons\n",
    "within the sequence. </span>\n",
    "\n",
    "**2**\n",
    "\n",
    "``` r\n",
    "paste(\"Lacy\", \"loves\", \"the\", \"Friday BIOS20172\", \"Lab\")\n",
    "```\n",
    "\n",
    "    [1] \"Lacy loves the Friday BIOS20172 Lab\"\n",
    "\n",
    "**3** <span style=\"color:blue\"> The line a \\<- 23:33 creates a vector a\n",
    "containing the numbers from 23:33 whereas the code b \\<- c(40, 30, 12,\n",
    "9, 27) creates a vector b with the specified values. The %in% operator\n",
    "is used to check whether each element of a is present in b. In other\n",
    "words, R compares each element of a against all the elements in b to\n",
    "determine if there is a match. The output will be a logical vector of\n",
    "the same length as a, where each element represents whether the\n",
    "corresponding element in a is found in b. For example, the first element\n",
    "of a is 23. When compared with all the elements of b, 23 is not found,\n",
    "so the first element of the output will be FALSE. So on and so forth.\n",
    "Each TRUE indicates that the corresponding element in a is found in b,\n",
    "while each FALSE indicates that the corresponding element in a is not\n",
    "present in b. </span>\n",
    "\n",
    "**4**\n",
    "\n",
    "``` r\n",
    "a <- c(\"xyz\")\n",
    "b <- letters\n",
    "\n",
    "a%in%b\n",
    "```\n",
    "\n",
    "    [1] FALSE\n",
    "\n",
    "**5**\n",
    "\n",
    "``` r\n",
    "for(i in 20:30){\n",
    "  if(i==28){\n",
    "    break\n",
    "  }\n",
    "  print(i)\n",
    "}\n",
    "```\n",
    "\n",
    "    [1] 20\n",
    "    [1] 21\n",
    "    [1] 22\n",
    "    [1] 23\n",
    "    [1] 24\n",
    "    [1] 25\n",
    "    [1] 26\n",
    "    [1] 27\n",
    "\n",
    "<span style=\"color:blue\"> 1. The loop starts with i being assigned 20\n",
    "(the first value in the sequence) </span>\n",
    "\n",
    "<span style=\"color:blue\"> 2. The condition i == 28 is checked. Since i\n",
    "is not equal to 28, the if statement condition evaluates to FALSE, and\n",
    "the code inside the if block is not executed. </span>\n",
    "\n",
    "<span style=\"color:blue\"> 3. The value of i (which is 20) is printed\n",
    "using the print() function. </span>\n",
    "\n",
    "<span style=\"color:blue\"> 4. The loop moves to the next iteration, and i\n",
    "is incremented to 21. The loop continues as previously stated, printing\n",
    "the values of i for each iteration until i reaches 28. When i becomes\n",
    "28, the condition i == 28 becomes TRUE, and the code inside the if block\n",
    "is executed. The break statement is encountered, which immediately exits\n",
    "the loop, stopping any further iterations. The output should list\n",
    "numbers 20 through and including 27. </span>\n",
    "\n",
    "<span style=\"color:blue\"> 5. The break function is responsible for\n",
    "prematurely ending the loop execution as soon as i becomes 28 (as per\n",
    "the condition: i == 28). </span>\n",
    "\n",
    "**6**\n",
    "\n",
    "``` r\n",
    "findORF <- function(seq) {\n",
    "  startcodon <- findStartCodons(seq)\n",
    "  stopcodon <- findStopCodons(seq)\n",
    "  \n",
    "  usedStop <- numeric(0)\n",
    "  ORFs <- character(0)\n",
    "  k <- 1\n",
    "  \n",
    "  for (start in startcodon) {\n",
    "    for (stop in stopcodon) {\n",
    "      if ((stop - start) %% 3 == 0 && stop > start) {\n",
    "        if (stop %in% usedStop) {\n",
    "          break\n",
    "        } else if (stop - start + 1 < 300) {\n",
    "          break\n",
    "        } else {\n",
    "          ORFs [k] <- paste(\"ORF starting at\", start, \"and ending at\", stop+2)\n",
    "          usedStop[k] <- stop\n",
    "            k <- k + 1\n",
    "            break\n",
    "          }\n",
    "        }\n",
    "      }\n",
    "    }\n",
    "  return(ORFs)\n",
    "}\n",
    "```\n",
    "\n",
    "**7**\n",
    "\n",
    "``` r\n",
    "setwd(\"C:/Users/15ull/OneDrive/Desktop/Labs - BIOS 20172 (Spring)/Lab5-BIOS20172_Spring2023\")\n",
    "\n",
    "library(seqinr)\n",
    "```\n",
    "\n",
    "    Warning: package 'seqinr' was built under R version 4.2.3\n",
    "\n",
    "``` r\n",
    "covid <- read.fasta(\"covid19.fasta\")[[1]]\n",
    "ORFs <- findORF(covid)\n",
    "print(ORFs)\n",
    "```\n",
    "\n",
    "    [1] \"ORF starting at 266 and ending at 13483\"  \n",
    "    [2] \"ORF starting at 13768 and ending at 21555\"\n",
    "    [3] \"ORF starting at 21536 and ending at 25384\"\n",
    "    [4] \"ORF starting at 25393 and ending at 26220\"\n",
    "    [5] \"ORF starting at 26523 and ending at 27191\"\n",
    "    [6] \"ORF starting at 27394 and ending at 27759\"\n",
    "    [7] \"ORF starting at 27894 and ending at 28259\"\n",
    "    [8] \"ORF starting at 28274 and ending at 29533\"\n",
    "\n",
    "**8**\n",
    "\n",
    "Yes, it is surprising that there are only 8 ORFs found considering the\n",
    "number of proteins coded by the virus. As per the article provided, the\n",
    "SARS-CoV-2 genomic sequences have 12 ORFs that encode 27 proteins. In\n",
    "the findORF function code, we limited results to ORFs with a minimum\n",
    "length of 100 amino acids. This could account for the 4 ORFs that aren’t\n",
    "reported from the covid19.fasta data.\n",
    "\n",
    "Protein isoform modification could allow the coronavirus genome to\n",
    "encode multiple proteins. More specifically, once proteins are formed,\n",
    "they can undergo further modifications, including phosphorylation,\n",
    "methylation, acetylation, etc. These modifications can alter the\n",
    "protein’s structure, function, and interaction with other proteins.\n",
    "Alternative splicing could also account for the multiple proteins\n",
    "encoded by the coronavirus genome.\n",
    "\n",
    "**9**\n",
    "\n",
    "library(seqinr)\n",
    "\n",
    "zika \\<- read.fasta(“zika.fasta”)\\[\\[1\\]\\] orfs \\<- findORF(zika)\n",
    "print(orfs)\n",
    "\n",
    "\\[1\\] “ORF starting at 107 and ending at 10366”\n",
    "\n",
    "**10**\n",
    "\n",
    "<span style=\"color:blue\"> Only one ORF was found using the findORF\n",
    "function. This is surprising given the relatively large size of the Zika\n",
    "virus genome (10,794 base pairs as per the article linked). According to\n",
    "the article, the entire Zika virus genome serves as both the viral\n",
    "genome and the mRNA. The genome is tranlsated into a single polyprotein\n",
    "that is 3,419 amino acids long. This polyprotein is then cleaved by both\n",
    "host and viral proteases, yielding all the structural and non-structal\n",
    "proteins required for viral replication. The strategy of having a single\n",
    "ORF and producing a polyprotein is a common feature among many\n",
    "positive-sense RNA viruses like the Zika virus. By encoding multiple\n",
    "proteins within a single ORF, positive-sense RNA viruses can maximize\n",
    "their coding capacity while minimizing the their genome size.\n",
    "Furthermore, this compact arrangement is advantageous as it allows for\n",
    "efficient packaging of genetic material within the viral particle.\n",
    "</span>\n",
    "\n",
    "**11**\n",
    "\n",
    "<span style=\"color:blue\"> Null Hypothesis: There is no association or\n",
    "relationship between the semester of birth and right-thumb or left-thumb\n",
    "preference. (In other words, the two are independent). Note: Wording was\n",
    "changed from quarter to semester sense the Q1/Q2 and Q3/Q4 columns had\n",
    "to be lumped to perform the chi square test in order to account for the\n",
    "values \\<5 in the expected table. </span> <span style=\"color:blue\">\n",
    "Alternative Hypotehsis: There is an association or relationship between\n",
    "the semester of birth and right-thumb or left-thumb preference. (In\n",
    "other words, the two are not independent). </span>\n",
    "\n",
    "**12**\n",
    "\n",
    "``` r\n",
    "x <- matrix(c(24.32,17.68,42,8.68,6.32,15,33,24,57), nrow = 3, byrow = TRUE)\n",
    "colnames(x) <- c(\"S1\", \"S2\", \"Totals\")\n",
    "rownames(x) <- c(\"Left-Thumb\", \"Right Thumb\", \"Totals\")\n",
    "print(x)\n",
    "```\n",
    "\n",
    "                   S1    S2 Totals\n",
    "    Left-Thumb  24.32 17.68     42\n",
    "    Right Thumb  8.68  6.32     15\n",
    "    Totals      33.00 24.00     57\n",
    "\n",
    "``` r\n",
    "T <- matrix(c(27,15,42, 6,9, 15, 33, 24, 57), nrow = 3, byrow = TRUE)\n",
    "colnames(T) <- c(\"S1\", \"S2\", \"Totals\")\n",
    "rownames(T) <- c(\"Left-Thumb\", \"Right Thumb\", \"Totals\")\n",
    "print(T)\n",
    "```\n",
    "\n",
    "                S1 S2 Totals\n",
    "    Left-Thumb  27 15     42\n",
    "    Right Thumb  6  9     15\n",
    "    Totals      33 24     57\n",
    "\n",
    "``` r\n",
    "a <- (T[3,1]/T[3,3]) %*% (T[1,3]/T[3,3]) %*% T[3,3]\n",
    "b <- (T[3,2]/T[3,3]) %*% (T[1,3]/T[3,3]) %*% T[3,3]\n",
    "c <- (T[3,1]/T[3,3]) %*% (T[2,3]/T[3,3]) %*% T[3,3]\n",
    "d <- (T[3,2]/T[3,3]) %*% (T[2,3]/T[3,3]) %*% T[3,3]\n",
    "\n",
    "t <- matrix(c(a,b,42,c,d,15,33,24,57), nrow = 3, byrow = TRUE)\n",
    "colnames(t) <- c(\"S1\", \"S2\", \"Totals\")\n",
    "rownames(t) <- c(\"Left-Thumb\", \"Right Thumb\", \"Totals\")\n",
    "print(t)\n",
    "```\n",
    "\n",
    "                       S1        S2 Totals\n",
    "    Left-Thumb  24.315789 17.684211     42\n",
    "    Right Thumb  8.684211  6.315789     15\n",
    "    Totals      33.000000 24.000000     57\n",
    "\n",
    "**13**\n",
    "\n",
    "``` r\n",
    "chi_squared <- function(O, E){\n",
    "  calculation <- sum((O-E)^2 / E)\n",
    "  return(calculation)\n",
    "}\n",
    "```\n",
    "\n",
    "**14**\n",
    "\n",
    "``` r\n",
    "Matrix1 <- matrix(c(27,15,6,9), nrow = 2, byrow = TRUE)\n",
    "Matrix2 <- matrix(c(24.32,17.68,8.68,6.32), nrow = 2, byrow = TRUE)\n",
    "\n",
    "O <- Matrix1\n",
    "E <- Matrix2\n",
    "result <- chi_squared(O, E)\n",
    "print(result)\n",
    "```\n",
    "\n",
    "    [1] 2.665494\n",
    "\n",
    "**15**\n",
    "\n",
    "``` r\n",
    "CI95 <- qchisq(p= 1-0.05, df = 1, ncp = 0, lower.tail = TRUE, log.p = FALSE)\n",
    "print(CI95)\n",
    "```\n",
    "\n",
    "    [1] 3.841459\n",
    "\n",
    "``` r\n",
    "CI99 <- qchisq(p= 1-0.01, df = 1, ncp = 0, lower.tail = TRUE, log.p = FALSE)\n",
    "print(CI99)\n",
    "```\n",
    "\n",
    "    [1] 6.634897\n",
    "\n",
    "``` r\n",
    "CI99.5 <- qchisq(p= 1-0.005, df = 1, ncp = 0, lower.tail = TRUE, log.p = FALSE)\n",
    "print(CI99.5)\n",
    "```\n",
    "\n",
    "    [1] 7.879439\n",
    "\n",
    "``` r\n",
    "CI95 <- qchisq(p= 1-0.05, df = 3, ncp = 0, lower.tail = TRUE, log.p = FALSE)\n",
    "print(CI95)\n",
    "```\n",
    "\n",
    "    [1] 7.814728\n",
    "\n",
    "``` r\n",
    "CI99 <- qchisq(p= 1-0.01, df = 3, ncp = 0, lower.tail = TRUE, log.p = FALSE)\n",
    "print(CI99)\n",
    "```\n",
    "\n",
    "    [1] 11.34487\n",
    "\n",
    "``` r\n",
    "CI99.5 <- qchisq(p= 1-0.005, df = 3, ncp = 0, lower.tail = TRUE, log.p = FALSE)\n",
    "print(CI99.5)\n",
    "```\n",
    "\n",
    "    [1] 12.83816\n",
    "\n",
    "<span style=\"color:blue\"> Note: When making the table of expected\n",
    "values, I lumped columns together since values less than 5 were present\n",
    "in multiple cells. For this question, I found the threshold chi-squared\n",
    "values for the grouped and ungrouped tables. The grouped table has two\n",
    "columns (semesters) and one degree of freedom. The ungrouped table has\n",
    "four columns (quarters) and three degrees of freedom. </span>\n",
    "\n",
    "<span style=\"color:blue\"> The chi squared value calculated in #14 is\n",
    "smaller than all of the threshold chi-squared values. Therefore, we fail\n",
    "to reject the null hypothesis at every significance level and conclude\n",
    "that there is no association between the semester of birth and\n",
    "right-thumb or left-thumb preference. The two are independent of one\n",
    "another. </span>\n",
    "\n",
    "**16**\n",
    "\n",
    "``` r\n",
    "Matrix1 <- matrix(c(27,15, 6,9), nrow = 2, byrow = TRUE)\n",
    "chisq.test(Matrix1)\n",
    "```\n",
    "\n",
    "        Pearson's Chi-squared test with Yates' continuity correction\n",
    "\n",
    "    data:  Matrix1\n",
    "    X-squared = 1.7707, df = 1, p-value = 0.1833\n",
    "\n",
    "``` r\n",
    "Matrix2 <- matrix(c(15,12,6,9,3,3,2,7),nrow=2, byrow = TRUE)\n",
    "chisq.test(Matrix2)\n",
    "```\n",
    "\n",
    "    Warning in chisq.test(Matrix2): Chi-squared approximation may be incorrect\n",
    "\n",
    "        Pearson's Chi-squared test\n",
    "\n",
    "    data:  Matrix2\n",
    "    X-squared = 3.688, df = 3, p-value = 0.2972\n",
    "\n",
    "<span style=\"color:blue\"> The result of this chi-squared test is similar\n",
    "to the result from #14 in the sense that we fail to reject the null\n",
    "hypothesis at every significance level. We conclude that there is no\n",
    "association between semester of birth and right-thumb or left-thumb\n",
    "preference. The p-value associated with the results of #14 is 0.1025,\n",
    "which is lower than the p-value of 0.1833 calculated here. However,\n",
    "p-values greater than 0.05 are not statistically significant and\n",
    "indicate strong evidence for the null hypothesis. So, in both cases, the\n",
    "p-value supports our failure to reject the null. Degrees of freedom for\n",
    "both tables are the same. Columns were lumped from quarters to semesters\n",
    "to account for values \\<5 such that the chi-squared test can be\n",
    "performed (rule of 5). </span>\n",
    "\n",
    "<span style=\"color:blue\"> Out of curiosity, I calculated the chi-squared\n",
    "value for the original observed table. This table violates the rule of 5\n",
    "when expected values are calculated. I wanted to see whether the code\n",
    "would still run. Regardless, the p-value exceeds 0.05 and provides\n",
    "evidence for the null hypothesis. The calculated chi-squared value is\n",
    "less than all threshold values calculated in #15. There are three\n",
    "degrees of freedom since four quarters are represented in the table\n",
    "(4-1).</span>\n",
    "\n",
    "**17**\n",
    "\n",
    "``` r\n",
    "bitter <- c(\"PTC Bitter\" = 10, \"PTC Not Bitter\" = 7)\n",
    "not <- c(\"PTC Bitter\" = 27, \"PTC Not Bitter\" = 32)\n",
    "kaleLeaf <- rbind(\"Plant Bitter\" = bitter, \"Plant Not Bitter\" = not)\n",
    "\n",
    "bitter <- c(\"PTC Not Bitter\" = 7, \"PTC Bitter\" = 8)\n",
    "not <- c(\"PTC Not Bitter\" = 32, \"PTC Bitter\" = 29)\n",
    "brusselSprouts <- rbind(\"Vegetable Not Bitter\" = not, \"Vegetable Bitter\" = bitter)\n",
    "\n",
    "bitter <- c(\"PTC Not Bitter\" = 5, \"PTC Bitter\" = 8)\n",
    "not <- c(\"PTC Not Bitter\" = 34, \"PTC Bitter\" = 29)\n",
    "broccoli <- rbind(\"Vegetable Not Bitter\" = not, \"Vegetable Bitter\" = bitter)\n",
    "\n",
    "bitter <- c(\"PTC Not Bitter\" = 3, \"PTC Bitter\" = 7)\n",
    "not <- c(\"PTC Not Bitter\" = 36, \"PTC Bitter\" = 30)\n",
    "rapiniStem <- rbind(\"Vegetable Not Bitter\" = not, \"Vegetable Bitter\" = bitter)\n",
    "\n",
    "bitter <- c(\"PTC Not Bitter\" = 7, \"PTC Bitter\" = 16)\n",
    "not <- c(\"PTC Not Bitter\" = 32, \"PTC Bitter\" = 21)\n",
    "rapiniLeaf <- rbind(\"Vegetable Not Bitter\" = not, \"Vegetable Bitter\" = bitter)\n",
    "\n",
    "bitter <- c(\"PTC Not Bitter\" = 26, \"PTC Bitter\" = 31)\n",
    "not <- c(\"PTC Not Bitter\" = 13, \"PTC Bitter\" = 6)\n",
    "arugulaLeaf <- rbind(\"Vegetable Not Bitter\" = not, \"Vegetable Bitter\" = bitter)\n",
    "\n",
    "print(kaleLeaf)\n",
    "```\n",
    "\n",
    "                     PTC Bitter PTC Not Bitter\n",
    "    Plant Bitter             10              7\n",
    "    Plant Not Bitter         27             32\n",
    "\n",
    "``` r\n",
    "print(brusselSprouts)\n",
    "```\n",
    "\n",
    "                         PTC Not Bitter PTC Bitter\n",
    "    Vegetable Not Bitter             32         29\n",
    "    Vegetable Bitter                  7          8\n",
    "\n",
    "``` r\n",
    "print(broccoli)\n",
    "```\n",
    "\n",
    "                         PTC Not Bitter PTC Bitter\n",
    "    Vegetable Not Bitter             34         29\n",
    "    Vegetable Bitter                  5          8\n",
    "\n",
    "``` r\n",
    "print(rapiniStem)\n",
    "```\n",
    "\n",
    "                         PTC Not Bitter PTC Bitter\n",
    "    Vegetable Not Bitter             36         30\n",
    "    Vegetable Bitter                  3          7\n",
    "\n",
    "``` r\n",
    "print(rapiniLeaf)\n",
    "```\n",
    "\n",
    "                         PTC Not Bitter PTC Bitter\n",
    "    Vegetable Not Bitter             32         21\n",
    "    Vegetable Bitter                  7         16\n",
    "\n",
    "``` r\n",
    "print(arugulaLeaf)\n",
    "```\n",
    "\n",
    "                         PTC Not Bitter PTC Bitter\n",
    "    Vegetable Not Bitter             13          6\n",
    "    Vegetable Bitter                 26         31\n",
    "\n",
    "**18**\n",
    "\n",
    "<span style=\"color:blue\"> A chi sqaured test can be used to test for a\n",
    "possible association between PTC bitterness and vegetable bitterness.\n",
    "None of the expected tables contain values less than 5 so the rule of 5\n",
    "isn’t violated. </span>\n",
    "\n",
    "**19**\n",
    "\n",
    "<span style=\"color:blue\"> Null Hypothesis: There is no relationship or\n",
    "association between the PTC bitterness and vegetable bitterness. In\n",
    "other words, the PTC bitterness is independent of the vegetable\n",
    "bitterness. </span> <span style=\"color:blue\"> Alternative Hypothesis:\n",
    "There is a relationship/association between the PTC bitterness and\n",
    "vegetable bitterness. In other words, the PTC and vegetable bitterness\n",
    "are not indepnedent of one another.</span>\n",
    "\n",
    "``` r\n",
    "bitter <- c(\"PTC Bitter\" = 9.39, \"PTC Not Bitter\" = 7.60)\n",
    "not <- c(\"PTC Bitter\" = 26.40, \"PTC Not Bitter\" = 32.61)\n",
    "kaleLeaf <- rbind(\"Plant Bitter\" = bitter, \"Plant Not Bitter\" = not)\n",
    "\n",
    "bitter <- c(\"PTC Not Bitter\" = 7.70, \"PTC Bitter\" = 7.30)\n",
    "not <- c(\"PTC Not Bitter\" = 31.30, \"PTC Bitter\" = 29.70)\n",
    "brusselSprouts <- rbind(\"Vegetable Not Bitter\" = not, \"Vegetable Bitter\" = bitter)\n",
    "\n",
    "bitter <- c(\"PTC Not Bitter\" = 6.67, \"PTC Bitter\" = 6.34)\n",
    "not <- c(\"PTC Not Bitter\" = 32.33, \"PTC Bitter\" = 30.67)\n",
    "broccoli <- rbind(\"Vegetable Not Bitter\" = not, \"Vegetable Bitter\" = bitter)\n",
    "\n",
    "bitter <- c(\"PTC Not Bitter\" = 5.13, \"PTC Bitter\" = 4.87)\n",
    "not <- c(\"PTC Not Bitter\" = 33.89, \"PTC Bitter\" = 32.13)\n",
    "rapiniStem <- rbind(\"Vegetable Not Bitter\" = not, \"Vegetable Bitter\" = bitter)\n",
    "\n",
    "bitter <- c(\"PTC Not Bitter\" = 11.80, \"PTC Bitter\" = 11.20)\n",
    "not <- c(\"PTC Not Bitter\" = 27.20, \"PTC Bitter\" = 25.80)\n",
    "rapiniLeaf <- rbind(\"Vegetable Not Bitter\" = not, \"Vegetable Bitter\" = bitter)\n",
    "\n",
    "bitter <- c(\"PTC Not Bitter\" = 29.25, \"PTC Bitter\" = 27.75)\n",
    "not <- c(\"PTC Not Bitter\" = 9.75, \"PTC Bitter\" = 9.25)\n",
    "arugulaLeaf <- rbind(\"Vegetable Not Bitter\" = not, \"Vegetable Bitter\" = bitter)\n",
    "\n",
    "print(kaleLeaf)\n",
    "```\n",
    "\n",
    "                     PTC Bitter PTC Not Bitter\n",
    "    Plant Bitter           9.39           7.60\n",
    "    Plant Not Bitter      26.40          32.61\n",
    "\n",
    "``` r\n",
    "print(brusselSprouts)\n",
    "```\n",
    "\n",
    "                         PTC Not Bitter PTC Bitter\n",
    "    Vegetable Not Bitter           31.3       29.7\n",
    "    Vegetable Bitter                7.7        7.3\n",
    "\n",
    "``` r\n",
    "print(broccoli)\n",
    "```\n",
    "\n",
    "                         PTC Not Bitter PTC Bitter\n",
    "    Vegetable Not Bitter          32.33      30.67\n",
    "    Vegetable Bitter               6.67       6.34\n",
    "\n",
    "``` r\n",
    "print(rapiniStem)\n",
    "```\n",
    "\n",
    "                         PTC Not Bitter PTC Bitter\n",
    "    Vegetable Not Bitter          33.89      32.13\n",
    "    Vegetable Bitter               5.13       4.87\n",
    "\n",
    "``` r\n",
    "print(rapiniLeaf)\n",
    "```\n",
    "\n",
    "                         PTC Not Bitter PTC Bitter\n",
    "    Vegetable Not Bitter           27.2       25.8\n",
    "    Vegetable Bitter               11.8       11.2\n",
    "\n",
    "``` r\n",
    "print(arugulaLeaf)\n",
    "```\n",
    "\n",
    "                         PTC Not Bitter PTC Bitter\n",
    "    Vegetable Not Bitter           9.75       9.25\n",
    "    Vegetable Bitter              29.25      27.75\n",
    "\n",
    "<span style=\"color:blue\"> I’m not sure what the question is referring to\n",
    "when it asks for PTC-curly kale stem data. I calculated the expected\n",
    "tables associated with the 5 observed tables provided in question 17. I\n",
    "asked a TA, but never heard back. The R function chisq.test is used in\n",
    "question 20 (I also asked a TA about this via email and was told I was\n",
    "fine to put the chisq.test under question 20). The only expected table\n",
    "that violates the rule of 5 is the table for the rapinistem. In this\n",
    "case, Fisher’s test may be more useful. All expected tables were\n",
    "calculated by hand. </span>\n",
    "\n",
    "**20**\n",
    "\n",
    "``` r\n",
    "chisq.test(kaleLeaf)\n",
    "```\n",
    "\n",
    "        Pearson's Chi-squared test with Yates' continuity correction\n",
    "\n",
    "    data:  kaleLeaf\n",
    "    X-squared = 0.24048, df = 1, p-value = 0.6239\n",
    "\n",
    "``` r\n",
    "chisq.test(brusselSprouts)\n",
    "```\n",
    "\n",
    "        Pearson's Chi-squared test with Yates' continuity correction\n",
    "\n",
    "    data:  brusselSprouts\n",
    "    X-squared = 5.1764e-32, df = 1, p-value = 1\n",
    "\n",
    "``` r\n",
    "chisq.test(broccoli)\n",
    "```\n",
    "\n",
    "        Pearson's Chi-squared test with Yates' continuity correction\n",
    "\n",
    "    data:  broccoli\n",
    "    X-squared = 5.4452e-30, df = 1, p-value = 1\n",
    "\n",
    "``` r\n",
    "chisq.test(rapiniStem)\n",
    "```\n",
    "\n",
    "    Warning in chisq.test(rapiniStem): Chi-squared approximation may be incorrect\n",
    "\n",
    "        Pearson's Chi-squared test with Yates' continuity correction\n",
    "\n",
    "    data:  rapiniStem\n",
    "    X-squared = 8.02e-31, df = 1, p-value = 1\n",
    "\n",
    "``` r\n",
    "chisq.test(rapiniLeaf)\n",
    "```\n",
    "\n",
    "        Pearson's Chi-squared test with Yates' continuity correction\n",
    "\n",
    "    data:  rapiniLeaf\n",
    "    X-squared = 2.3831e-31, df = 1, p-value = 1\n",
    "\n",
    "``` r\n",
    "chisq.test(arugulaLeaf)\n",
    "```\n",
    "\n",
    "        Pearson's Chi-squared test\n",
    "\n",
    "    data:  arugulaLeaf\n",
    "    X-squared = 0, df = 1, p-value = 1\n",
    "\n",
    "<span style=\"color:blue\"> Every test except for the chi-squared test\n",
    "performed on the RapiniLeaf table provide evidence in support of the\n",
    "null hypothesis since p-values exceed 0.05 (which suggests that the\n",
    "results are not statistically significant). Support of the null\n",
    "hypothesis indicates that there is no association between PTC and\n",
    "vegetable bitterness. The p-value calculated for the RapiniLeaf table is\n",
    "0.0316, which is less than 0.05. </span>\n",
    "\n",
    "<span style=\"color:blue\"> The threshold chi-squared value for one degree\n",
    "of freedom and a significance level of 0.05 is approximately 3.84. All\n",
    "calculated chi-squared values are less than 3.84 except for the\n",
    "chi-squared value calculated for the RapiniLeaf. Therefore, we would\n",
    "fail to reject the null hypothesis in every case except for the\n",
    "RapiniLeaf. In the case of the RapiniLeaf, we reject the null and\n",
    "conclude that there is an association between PTC and vegetable\n",
    "bitterness. One degree of freedom is used for every contingency table\n",
    "since they all contain two columns (2-1). </span>\n",
    "\n",
    "**21**\n",
    "\n",
    "<span style=\"color:blue\"> The data largely demonstrates no association\n",
    "between PTC and vegetable bitterness. Biologically, this could be\n",
    "attributed to several factors including the following: genetic\n",
    "variation, tast perception thresholds, compensatory taste mechanisms,\n",
    "and psychological factors. </span>\n",
    "\n",
    "<span style=\"color:blue\"> Genetic Variation: Individuals may have\n",
    "genetic variations that affect their ability to taste bitterness.\n",
    "Genetic factors can influence taste receptor genes and their sensitivity\n",
    "to bitter compounds. Therefore, even if vegetables contain varying\n",
    "levels of bitterness, some individuals may not be able to perceive it\n",
    "due to their genetic makeup.</span>\n",
    "\n",
    "<span style=\"color:blue\"> Taste Perception Thresholds: Taste perception\n",
    "thresholds can vary among individuals. Some individuals may have higher\n",
    "or lower thresholds for perceiving bitterness. Even if a vegetable is\n",
    "objectively bitter, individuals with higher thresholds may not perceive\n",
    "it as strongly, or vice versa. </span>\n",
    "\n",
    "<span style=\"color:blue\"> Compensatory Taste Mechanisms: The human sense\n",
    "of taste is complex, and our perception of taste is influenced by\n",
    "various factors. Compensatory taste mechanisms, such as the presence of\n",
    "other flavors or textures in a meal, can modulate our perception of\n",
    "bitterness. It is possible that individuals may find ways to mask or\n",
    "counteract the bitterness of vegetables through other sensory cues,\n",
    "making the perceived association between ptc and vegetable bitterness\n",
    "weaker. </span>\n",
    "\n",
    "<span style=\"color:blue\"> Psychological Factors: Psychological factors,\n",
    "such as expectations and preferences, can also influence taste\n",
    "perception. If individuals have preconceived notions or strong\n",
    "preferences for certain vegetables, it may affect their perception of\n",
    "bitterness. For example, individuals who enjoy a particular vegetable\n",
    "may perceive it as less bitter than individuals who dislike it,\n",
    "regardless of its actual bitterness level. </span>"
   ]
  }
 ],
 "nbformat": 4,
 "nbformat_minor": 5,
 "metadata": {}
}
